tests/bot/cogs/test_token_remover.py

```diff
@@ -1,131 +1,279 @@
-import asyncio
-import logging
 import unittest
-from unittest.mock import AsyncMock, MagicMock
+from unittest import mock
+from unittest.mock import MagicMock
 
 from discord import Colour
 
-from bot.cogs.token_remover import (
-    DELETION_MESSAGE_TEMPLATE,
-    TokenRemover,
-    setup as setup_cog,
-)
-from bot.constants import Channels, Colours, Event, Icons
-from tests.helpers import MockBot, MockMessage
+from bot import constants
+from bot.cogs import token_remover
+from bot.cogs.moderation import ModLog
+from bot.cogs.token_remover import TokenRemover
+from tests.helpers import MockBot, MockMessage, autospec
 
 
-class TokenRemoverTests(unittest.TestCase):
+class TokenRemoverTests(unittest.IsolatedAsyncioTestCase):
     """Tests the `TokenRemover` cog."""
 
     def setUp(self):
         """Adds the cog, a bot, and a message to the instance for usage in tests."""
         self.bot = MockBot()
-        self.bot.get_cog.return_value = MagicMock()
-        self.bot.get_cog.return_value.send_log_message = AsyncMock()
         self.cog = TokenRemover(bot=self.bot)
 
-        self.msg = MockMessage(id=555, content='')
-        self.msg.author.__str__ = MagicMock()
-        self.msg.author.__str__.return_value = 'lemon'
-        self.msg.author.bot = False
-        self.msg.author.avatar_url_as.return_value = 'picture-lemon.png'
-        self.msg.author.id = 42
-        self.msg.author.mention = '@lemon'
+        self.msg = MockMessage(id=555, content="hello world")
         self.msg.channel.mention = "#lemonade-stand"
+        self.msg.author.__str__ = MagicMock(return_value=self.msg.author.name)
+        self.msg.author.avatar_url_as.return_value = "picture-lemon.png"
 
-    def test_is_valid_user_id_is_true_for_numeric_content(self):
-        """A string decoding to numeric characters is a valid user ID."""
-        # MTIz = base64(123)
-        self.assertTrue(TokenRemover.is_valid_user_id('MTIz'))
+    def test_is_valid_user_id(self):
+        """Should correctly discern valid user IDs and ignore non-numeric and non-ASCII IDs."""
+        subtests = (
+            ("MTIz", True),  # base64(123)
+            ("YWJj", False),  # base64(abc)
+            ("λδµ", False),
+        )
 
-    def test_is_valid_user_id_is_false_for_alphabetic_content(self):
-        """A string decoding to alphabetic characters is not a valid user ID."""
-        # YWJj = base64(abc)
-        self.assertFalse(TokenRemover.is_valid_user_id('YWJj'))
+        for user_id, is_valid in subtests:
+            with self.subTest(user_id=user_id, is_valid=is_valid):
+                result = TokenRemover.is_valid_user_id(user_id)
+                self.assertIs(result, is_valid)
```

Comment:
I'm not sure that I agree with the usage of `assertIs()` in this case rather than `assertEqual()`. Although in this case the result is the same, you typically want to use `assertIs()` to specifically convey that you're comparing object identity, which isn't relevant in this case.

Reply:
Also, this may be more of a personal preference, but I've found that it's significantly easier to read and debug when the valid and invalid cases are separated into 2+ tests, and then using something like `assertTrue(result)` and `assertFalse(result)` for each of the subtests respectively. At a glance, it can sometimes be needlessly difficult to discern whether it should be true or false for that specific input, especially if the list of test inputs grow.

Reply:
Well you wouldn't consider `if x == True:` correct, right? So why would `assertEqual` be correct when comparing booleans? Sure it works but I've learned it's not good practice to compare booleans and `None` with `==`.

Reply:
I agree on the readability point but I didn't feel it was worth splitting up given the low amount of test inputs and the simplicity of the test. If more inputs are added like you suggested, then I'll probably split it up.

Reply:
> Well you wouldn't consider if x == True: correct, right? So why would assertEqual be correct when comparing booleans? Sure it works but I've learned it's not good practice to compare booleans and None with ==.

Yeah, disregard that comment; you're correct. It's actually better to use `assertIs()` over `assertEqual()` when directly testing a boolean return, since the truthiness of the object (from `bool(object)`) could give rather misleading results if the function returns anything other than `True`, `False`, or `None` when it shouldn't. My preference is generally towards `assertTrue()` or `assertFalse()`, but in this situation `assertIs()` is fine.

Sorry about that, I'm honestly not quite sure what I was thinking there. I've been just getting over an ear infection and may been a bit too eager to get back into PR review. I'll take this as a sign to take a few more days off before doing anything that I don't need to address immediately. ;-)

Reply:
No worries. I hope you don't get discouraged; your reviews would be valuable here.